From c93382c44afa52f781bcb45f0b76da2b9427decd Mon Sep 17 00:00:00 2001
From: Xianzhong <xianzhong.li@nxp.com>
Date: Sat, 17 Nov 2018 02:19:02 +0800
Subject: [PATCH 5148/5242] MGS-4363 [#imx-1246] fix coverity resource leak
 issue

commit  01c6a1b062e79a4e41bd3618ddcb6778c0ea4159 from
https://source.codeaurora.org/external/imx/linux-imx.git

CID 3925877: Resource leak (RESOURCE_LEAK)
14. leaked_storage: Variable pages going out of scope leaks the storage it points to.

Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
(cherry picked from commit a38d1b1ff31ce87a231588de1a2e24a787f87143)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../default/gc_hal_kernel_allocator_user_memory.c  |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_user_memory.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_user_memory.c
index 3a04df6..7d4c378 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_user_memory.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_user_memory.c
@@ -214,9 +214,9 @@ static int import_page_map(struct um_desc *um,
     kfree(um->sgt.sgl);
 #endif
 
-    if (um->pages)
+    if (pages)
     {
-        kfree(um->pages);
+        kfree(pages);
     }
     return result;
 }
-- 
1.7.9.5

